### ISSUE TITLE
Invite URL redirects to login page

### ISSUE THREAD

**Issue by fpapadamis** [Infisical/infisical#1774]
### Describe the bug
When inviting new members to the organization, the generated URL redirects to the login page, thus the user invited is not able to create an account. This happens when using email invitation as well as direct URL invitations.

### To Reproduce
1. Go to `Access Control`.
2. Click on '+ Add Member`.
3. FIll in the user's email address.
4. Click on `Add Member`.
5. Use the URL provided in the email to create an account or, if SMTP is not configured, use the generated URL directly.

### Expected behavior
When the user browses the URL, it is expected that they will be able to create a new account by entering their password etc.

### Platform you are having the issue on:
Self-hosted Infisical using docker compose.

**Comment by akhilmhdh** [comment]
Did you set the SITE_URL environment variable

**Comment by fpapadamis** [comment]
Yes, I have set the `SITE_URL` environment variable. The URL is generated correctly.

**Comment by akhilmhdh** [comment]
Hey @fpapadamis 

I have tested the workflow and it seems to be working. The login page pops up when you have already an account. Are you sure the given email as never been used. 

| Removing from org or doesn't remove the user.

**Comment by fpapadamis** [comment]
Hi @akhilmhdh, I have tested it with email addresses that have never been used and with addresses that have been used and have been removed. They both redirect to the login page. This bug has also been brought to attention in the Slack channel from other members.

**Comment by akhilmhdh** [comment]
Have you by chance disabled signups in your admin panel @fpapadamis

**Comment by fpapadamis** [comment]
Sorry @akhilmhdh, I forgot to include that signups are disabled.
I am refering to [this](https://infisical-users.slack.com/archives/C04BSBMQAQ7/p1713438900430369) thread in Slack.
I am updating the issue description.

**Comment by akhilmhdh** [comment]
Ahh @fpapadamis Ok so that's why its being redirect to the login.

You need to allow signup and then inside `Restrict signup by email domain(s)` you need to configure the email you want to restrict to control invitation
